pkg/util/validation/validate.go

```diff
@@ -100,6 +120,38 @@ func ValidateSample(cfg SampleValidationConfig, userID string, ls []cortexpb.Lab
 	return nil
 }
 
+func ValidateExemplar(userID string, ls []cortexpb.LabelAdapter, e cortexpb.Exemplar) ValidationError {
+	if len(e.Labels) <= 0 {
+		DiscardedExemplars.WithLabelValues(exemplarLabelsMissing, userID).Inc()
+		return fmt.Errorf(`exemplar missing labels: series: %s`,
+			cortexpb.FromLabelAdaptersToLabels(ls).String())
+	}
+
+	if e.TimestampMs == 0 {
```

Comment:
[question] Why is 0 invalid?

Reply:
Technically an exemplar will always have a timestamp. If it is not exported with one (usually it would be) Prometheus scrape assigns the scrape timestamp. So, any exemplar coming into Cortex via remote write will should a timestamp. I think checking for 0 is all we can do without changing the timestamp field to a pointer in the proto?

Reply:
> I think checking for 0 is all we can do without changing the timestamp field to a pointer in the proto?

Yes, correct. It's fine to me.